code/modules/surgery/amputation.dm

```diff
@@ -34,6 +33,9 @@
 	success_sound = 'sound/surgery/organ2.ogg'
 
 /datum/surgery_step/sever_limb/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	if(HAS_TRAIT(target, TRAIT_NODISMEMBER))
+		to_chat(user, span_warning("You fail to amputate [target]'s [parse_zone(target_zone)]!"))
+		return SURGERY_STEP_FAIL
```

Comment:
Someone's going to get pretty far into the surgery only to be disappointed if you don't add this check to
`/datum/surgery/amputation/can_start(mob/user, mob/living/patient)` as well

Reply:
i think its preferrable to have it here so you cant just use drapes to know whether someone has TRAIT_NODISMEMBER from sleeping carp

Reply:
`As in, add a can_start proc to /datum/surgery/amputation` and return false if nodismember. That will make it not even show up in the list of possible surgeries!

Reply:
that literally does what i just said previously

Reply:
it is done